### DISCUSSION TITLE
GitHub copilot is not working on the remote server

### DISCUSSION THREAD

**Discussion by abhijitm08** [community/community#50328] [Copilot Conversations]
### Select Topic Area

Question

### Body

The GitHub Copilot does not seem to work on remote SSH, however does locally. I started having this problem after an update of VS code yesterday and not before. Strangely though, the GitHub Copilot Labs does work. Any help is appreciated! 

I am using the following VS code version. 

```
Version: 1.76.2
Commit: ee2b180d582a7f601fa6ecfdad8d9fd269ab1884
Date: 2023-03-14T17:54:09.061Z
Electron: 19.1.11
Chromium: 102.0.5005.196
Node.js: 16.14.2
V8: 10.2.154.26-electron.0
OS: Darwin x64 22.3.0
Sandboxed: No
```

**Comment by Anmol-Baranwal** [comment]
If GitHub Copilot is not working on remote SSH but is working locally, it could be due to a few possible reasons. Here are some steps you can try to troubleshoot the issue:

1. Ensure that you have installed and enabled the GitHub Copilot extension on both your local and remote machines.

2. Check that the SSH connection to the remote machine is working properly. You can try running basic commands on the remote machine to see if they work as expected.

3. Make sure that the remote machine has the necessary dependencies installed to run GitHub Copilot. You can check the GitHub Copilot documentation for the specific requirements.

4. Try updating both VS Code and the GitHub Copilot extension on your local and remote machines to the latest versions.

5. If the issue persists, try disabling any other extensions that might interfere with GitHub Copilot's operation, as they could be conflicting with it.

> These are general steps. Please let me know if they resolve your issues. Sometimes updates can mess up unnecessary things."

**Reply by abhijitm08** [reply]
1) They are enabled globally. 

Local: 
![Screenshot 2023-03-17 at 17 37 52](https://user-images.githubusercontent.com/9482011/225965365-cd069916-ab77-47ed-be10-5c6bd6c9d260.png)

Remote: 

![Screenshot 2023-03-17 at 17 37 24](https://user-images.githubusercontent.com/9482011/225965275-6c4d0462-7cf9-4af7-9801-2eea006359cb.png)

2. Everything works okay on the remote server just on the terminal. 

3. The copilot was working yesterday on the remote server, it is after the update I am getting this error. 

4. As you can see from the images above copilot is up-to-date and the VS code version is 1.76.2 which is also the latest version. 

5. The only extension I can think of that would interfere is GitHub Copilot Lab and I have disabled it, but the problem persists. Here is the list of all my extensions. 

![Screenshot 2023-03-17 at 17 44 34](https://user-images.githubusercontent.com/9482011/225967067-10e77983-2af5-427d-80c8-f4876c14ccae.png)

**Comment by Mohamed-SayedAlAhl** [comment]
It seems like you have already tried some troubleshooting steps, but the issue with GitHub Copilot not working on remote SSH is still persisting. Here are a few additional steps you can try:

1. Check if the remote server has the required dependencies installed for GitHub Copilot to work. You can refer to the GitHub Copilot documentation for the specific requirements.

2. Try running VS Code with the `--disable-extensions` flag on the remote server to see if any other extension is conflicting with GitHub Copilot. You can do this by opening the terminal on the remote server and running the following command:

   ```
   code --disable-extensions
   ```

   If GitHub Copilot works with this command, you can try disabling other extensions one by one to identify the conflicting extension.

3. If you are using an older version of the OpenSSH client, try updating it to the latest version as GitHub Copilot requires a version of OpenSSH that supports the `-T` flag.

4. Check if there are any firewall rules or network restrictions on the remote server that might be preventing GitHub Copilot from connecting to the necessary resources.

5. Try uninstalling and reinstalling the GitHub Copilot extension on both your local and remote machines.

If none of the above steps help, you can try reaching out to the GitHub support team for further assistance.

**Reply by botcs** [reply]
Hi @Mohamed-SayedAlAhl ,

I got this message:
```
$ code --disable-extensions
Ignoring option 'disable-extensions': not supported for code.
```

**Reply by Mohamed-SayedAlAhl** [reply]
Hi @botcs,

It looks like the `--disable-extensions` flag is not supported for the `code` command. You can try a different approach to disable extensions in VS Code when running it on a remote server.

One way to do this is to add `"extensions.enabled": false` to your VS Code settings for the remote server. You can do this by following these steps:

1. Open VS Code on the remote server.
2. Press `Ctrl + ,` to open the Settings editor.
3. In the search bar at the top, type "extensions.enabled" and press Enter.
4. Under "Remote", check the "Edit in settings.json" button.
5. Add the following line to your settings.json file: `"extensions.enabled": false`
6. Save the file and restart VS Code.

With this setting, VS Code should start without any extensions enabled, allowing you to test if GitHub Copilot works on the remote server. If it does, you can try enabling extensions one by one to see if any of them are causing the conflict.

**Comment by abhijitm08** [comment]
Hi, I removed `rm -rf .vscode-server` on remote server and removed all plugins on remote. Then logged in fresh with VS-code and installed minimal set of plugins needed. It is all fine now. So closing the issue.

**Reply by botcs** [reply]
Hi, @abhijitm08 

Removing the .vscode-server didn't help for me. It seems like that on the server side every evidence of existing copilot installation is missing.

Client is working fine, proxyJumps are working fine, as this happens only on one specific server.

No debugging output can be found (Copilot is missing from the OUTPUT pages). Really don't know where to look.

Any suggestions @Mohamed-SayedAlAhl ? Thanks a lot for looking into this! :)

**Comment by jgrimard** [comment]
On the bottom left side of VS Code, click on the Accounts button.  Make sure you turn ON "remote tunnel access" or copilot won't work in the remote session.  That fixed it for me.

**Reply by AntonBogun** [reply]
This only works if you have access to VSCode on the host machine and aren't already in a remote.

**Comment by AntonBogun** [comment]
This issue is very much still a thing. Turning on "remote tunnel access" starts up a vscode host on your machine, not the actual server.
I am noticing no output whatsoever in the Output>GitHub Copilot, but I keep getting errors in "Output>Extension Host (Remote)" that look like this every time Copilot is ought to be providing suggestions:
```
2023-07-13 21:57:02.633 [error] [GitHub.copilot] provider FAILED
2023-07-13 21:57:02.633 [error] RangeError: WebAssembly.Memory(): could not allocate memory
    at /users/***/.vscode-server/extensions/github.copilot-1.96.259/dist/worker.js:1:112788
    at new Promise (<anonymous>)
    at Function.init (/users/***/.vscode-server/extensions/github.copilot-1.96.259/dist/worker.js:1:108161)
    at /users/***/.vscode-server/extensions/github.copilot-1.96.259/dist/worker.js:1:50581
    at E (/users/***/.vscode-server/extensions/github.copilot-1.96.259/dist/worker.js:1:50683)
    at v (/users/***/.vscode-server/extensions/github.copilot-1.96.259/dist/worker.js:1:50748)
    at u.isEmptyBlockStart (/users/***/.vscode-server/extensions/github.copilot-1.96.259/dist/worker.js:1:57032)
    at Object.t.isEmptyBlockStart (/users/***/.vscode-server/extensions/github.copilot-1.96.259/dist/worker.js:1:64005)
    at MessagePort.<anonymous> (/users/***/.vscode-server/extensions/github.copilot-1.96.259/dist/worker.js:1:181106)
    at MessagePort.[nodejs.internal.kHybridDispatch] (node:internal/event_target:694:20)
```

**Comment by HaoLiuHust** [comment]
similar problem, I can use copilot chat in local, but hard to use in remote, it always "thinking"

**Comment by LongQIByte** [comment]
I am experiencing a similar problem. However, there is a difference in my case: the Copilot feature returns to normal if Visual Studio Code (VSCode) is left open for more than 10 minutes.

```
[INFO] [ghostText] [2023-11-05T12:56:23.108Z] Cancelled after extractPrompt
[INFO] [ghostText] [2023-11-05T12:56:23.108Z] Cancelled after extractPrompt
[INFO] [ghostText] [2023-11-05T12:56:23.109Z] Cancelled after extractPrompt
[INFO] [ghostText] [2023-11-05T12:56:23.111Z] Cancelled after extractPrompt
[INFO] [ghostText] [2023-11-05T12:56:23.112Z] Cancelled after extractPrompt
[INFO] [ghostText] [2023-11-05T12:58:56.588Z] Cancelled after requestMultiline
[INFO] [default] [2023-11-05T13:00:35.603Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2023-11-05T13:00:36.318Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 715 ms
[INFO] [streamChoices] [2023-11-05T13:00:36.320Z] solution 0 returned. finish reason: [stop]
[INFO] [streamChoices] [2023-11-05T13:00:36.323Z] request done: headerRequestId: [e621d60a-19e0-48ec-b6ad-321ae8547de8] model deployment ID: [xca6373c2f5d7]
```

**Comment by marcoBmota8** [comment]
Hi I am experiencing a related but not exactly the same issue.

For me the chat works perectly on both local and remote machine. However the inline siuggestions are completely gone in the remote machine although they work fine in my local machine.

Any troubleshooting for this scenario?

**Reply by GoldenRain-Su** [reply]
Hi, I met the same problem. Do you solve it?

**Comment by zoffixznet** [comment]
Wanted to add: had similar issue with Copilot icon having crossed out line through it on remote servers only. I did `rm -rf ~/.vscode-server` which didn't help anything.

Then in Github Copilot settings, under "Github>Copilot: Enable" it was set to false for "\*" and only markdown and plaintext were set to true. Changing "\*" to true (or adding individual langs) fixed the issue.

I've no idea how this setting got changed that way. Copilot used to work fine for me before.

**Reply by marcoBmota8** [reply]
For me the copilot symbol is not crossed out. It is enabled globally. In fact the sidebar chat works perfectly. It is just the inline suggestions that are not popping. I checked all settings for inloine suggestions and everything is exactly as in the local scenario. There is something going on with the SSH remote extension that blocks that particular feature for some reason.

**Comment by lloydtripp** [comment]
I'm also having this issue. Github co-pilot works locally but I get an "Out of Memory" issue when remote ssh to a server. 

![image](https://github.com/community/community/assets/6070112/4f9adc18-c0fd-4d7e-a80c-cb7fe5da3234)


This is the output from GitHub Copilot:
```
[INFO] [code-referencing] [2024-01-24T22:49:49.717Z] Public code references are enabled.
[INFO] [ghostText] [2024-01-24T22:49:50.717Z] Cancelled after extractPrompt
[INFO] [ghostText] [2024-01-24T22:49:50.718Z] Cancelled after extractPrompt
[INFO] [ghostText] [2024-01-24T22:49:51.080Z] Cancelled after extractPrompt
```
[Copilot Diagnostics.md](https://github.com/community/community/files/14044480/Copilot.Diagnostics.md)

**Comment by lloydtripp** [comment]
See this solution

Add this to your VSCode settings:

    "remote.extensionKind": {
        "GitHub.copilot": [
            "ui",
        ],
    }
See https://github.com/orgs/community/discussions/6942#discussioncomment-4962858.

https://github.com/orgs/community/discussions/15307#discussioncomment-5271054

Worked for me.

**Comment by ilgatto88** [comment]
I had the same issue, but solved it via changing debug.useNodeFetcher to false in the advanced settings:
` 
"github.copilot.advanced": {
    
        "debug.useElectronFetcher": true,
        "debug.useNodeFetcher": false,
    },
`

**Comment by pwahs** [comment]
I had a different problem which will probably only affect the minority, but maybe it helps someone. I could log in, but the chat view would keep showing the "Sign in" button instead of my chat history. In the Output view, set to the "GitHub Copilot Chat" channel, I saw this:

[info] Logged in as ...my github user name...
[error] GitHub Copilot could not connect to server. Extension activation failed: "fetch failed"

Solution: I had to disconnect the VPN connection that I was using for unrelated work and still had running. Then I could log in as intended.
